### PR TITLE
#323 Email service

### LINKED ISSUE #323: Email sending service
Task list:
- Research could we use mimirapp.xyz domain for sending emails
- Choose appropriate technical stack for the task
- Create a Nest Service for emails

_Acceptance criteria_
- Email sending does not block execution flow (should be async)

### PR DESCRIPTION
**Pull request by x8er**
closes #323

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,6 +125,7 @@
     "i18next-http-backend": "^1.4.0",
     "lodash": "^4.17.21",
     "metro-config": "0.70.1",
+    "nodemailer": "^6.7.8",
```
Comment (unrealsolver) [inline package.json:128]: Can we use a single version?

File: packages/email-service/src/app/app.service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return { message: 'Welcome to email-service!' };
+  }
+
+  async sendOverdueNotify(receivers: string[]): Promise<boolean> {
+    try {
+      await this.transporter.sendMail({
+        from: '"Mimir App" <app@mimirapp.xyz>',
+        to: receivers.join(', '),
+        subject: 'Overdue books | Mimir App',
+        text: 'Test message',
```
Comment (unrealsolver) [inline packages/email-service/src/app/app.service.ts:25]: we can already make it more generic, by creating an `Email` type, which will encapsulate all email-related data (body, from, to, subj, etc). And email service should know as little as possible about books, overdues and donations.

File: packages/email-service/src/app/app.controller.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  @Get()
+  getData() {
+    return this.appService.getData();
+  }
+
+  @Post('/send-overdue-notify')
+  async sendOverdueNotify(@Body() receivers: string[], @Res() res: Response) {
+    if (!receivers.length) return res.status(404).send();
+    const sentOk = await this.appService.sendOverdueNotify(receivers);
```
Comment (unrealsolver) [inline packages/email-service/src/app/app.controller.ts:17]: Generally, would approve your approach of having a separate service. But let's consider following changes - there is should be a some queue between BE app and a mail service. So BE could post emails to the queue and instantly receive 200 OK. Email service pulls emails from that queue and sends them in order. The queue could be part of email-service. (e.g. https://github.com/bee-queue/bee-queue)

**Review by unrealsolver — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (x8er) [reply package.json:128]: yeah, why not

**Review by sabrina34352 — APPROVED** [review]

**Pull request merged**